tests/test_skills/test_apy_estimation_abci/test_behaviours.py

```diff
@@ -2639,68 +2635,172 @@ def test_reset_behaviour(
+        ),
+    )
+    def test_reset_behaviour(
         self,
-        monkeypatch: MonkeyPatch,
-        no_action: Callable[[Any], None],
         caplog: LogCaptureFixture,
+        is_most_voted_estimate_set: bool,
+        estimations: pd.DataFrame,
+        is_broadcasting_to_server: bool,
+        log_level: int,
+        log_message: str,
     ) -> None:
-        """Test reset behaviour without most voted estimate."""
+        """Test reset behaviour."""
         self.fast_forward_to_behaviour(
             behaviour=self.behaviour,
             behaviour_id=self.behaviour_class.behaviour_id,
             synchronized_data=SynchronizedData(
-                AbciAppDB(setup_data=dict(most_voted_estimate=[None]))
+                AbciAppDB(
+                    setup_data=dict(
+                        most_voted_estimate=["not_None"]
+                        if is_most_voted_estimate_set
+                        else [None],
+                        period_count=[0],
+                    )
+                )
             ),
         )
-        behaviour = cast(BaseBehaviour, self.behaviour.current_behaviour)
+        behaviour = cast(BaseResetBehaviour, self.behaviour.current_behaviour)
         assert behaviour.behaviour_id == self.behaviour_class.behaviour_id
 
-        monkeypatch.setattr(BenchmarkTool, "save", lambda _: no_action)
-        monkeypatch.setattr(AbciApp, "last_timestamp", datetime.now())
+        # we do this because of https://github.com/valory-xyz/open-autonomy/pull/646
+        behaviour._get_finalized_estimates = lambda: estimations  # type: ignore
+        behaviour._send_to_server = mock.MagicMock()  # type: ignore
 
-        self.behaviour.context.params.observation_interval = 0.1
+        behaviour.params.is_broadcasting_to_server = is_broadcasting_to_server
+        behaviour.params.observation_interval = SLEEP_TIME_TWEAK
 
         with caplog.at_level(
-            logging.INFO,
+            log_level,
             logger="aea.test_agent_name.packages.valory.skills.apy_estimation_abci",
         ):
-            self.behaviour.act_wrapper()
-            cast(
-                CycleResetBehaviour, self.behaviour.current_behaviour
-            ).params.sleep_time = SLEEP_TIME_TWEAK
-            time.sleep(SLEEP_TIME_TWEAK + 0.01)
-            self.behaviour.act_wrapper()
+            behaviour.act_wrapper()
 
-        assert "[test_agent_name] Entered in the 'cycle_reset' behaviour" in caplog.text
-        assert (
-            "[test_agent_name] Finalized estimate not available. Resetting!"
-            in caplog.text
-        )
+        if log_message:
+            assert log_message in caplog.text
+
+        if (
+            is_broadcasting_to_server
+            and is_most_voted_estimate_set
+            and estimations is not None
+        ):
+            behaviour._send_to_server.assert_called_once_with(estimations)
+
+        time.sleep(SLEEP_TIME_TWEAK + 0.01)
```

Comment:
why double?

Reply:
It is named `SLEEP_TIME_TWEAK` because we tweak the behaviour's sleep time with its value above:

https://github.com/valory-xyz/open-autonomy/blob/e5672c3cef9115bfa09a8671ee801d72cdf9ba02/tests/test_skills/test_apy_estimation_abci/test_behaviours.py#L2784

We do not double it, we just add some tiny tolerance. This should be a constant though.